lib/internal/loader/resolveRequestUrl.js

```diff
@@ -80,6 +80,10 @@ const resolveRequestUrl = (baseURLOrString, specifier) => {
   const baseURL = normalizeBaseURL(baseURLOrString);
   let url = search(specifier, baseURL);
 
+  if (url.protocol === 'node:' && NativeModule.nonInternalExists(url.pathname)) {
```

Comment:
@bmeck This was needed to support `import('fs')`. The problem was that the URL this function produces (`node:fs`) wasn't accepted as an input, leading to fragility ("better not try to normalize URLs twice!"). With this change, normalizing an already normalized URL *should* be safe.

Reply:
can you explain this a bit more, right now `import("node:fs")` should throw an error from user code.

Reply:
Gotcha. The general problem is that I resolve the url before calling `loader.import` and as a side effect `node:fs` makes it into first level of importing. I think I can rewrite that to never pass through `loader.import` or, alternatively, make `loader.import` take a different referrer/base url. Which would you prefer?